arch/arm/src/sama5/Kconfig

```diff
@@ -799,6 +799,7 @@ config SAMA5_ARM
 config SAMA5_FUSE
 	bool "Fuse Controller (FUSE)"
 	default n
+	depends on SAMA5_HAVE_FUSE
```

Comment:
@TimJTi is there some SAMA5 MCU that doesn't have FUSE? Maybe this "depends on" is not needed

Reply:
> @TimJTi is there some SAMA5 MCU that doesn't have FUSE? Maybe this "depends on" is not needed

@acassis - SAMA5D2 and D4 have "secure fuse controller" and SAMA5D3 has just a "fuse controller", and they do all seem to have one or the other. What is missing from Kconfig is selecting SAMA5_FUSE for the ARCH_CHIP_SAMA5D3 family devices. So it is needed if anyone ever wants to add a non-secure Fuse Controller driver for that device. It used to be set for A2 and A4, but that was "wrong" in my mind as they have SFC instead so I add that instead/

I can add it for the SAMA3 completeness if wanted, even though it won't actually enable anything useful?

Reply:
@TimJTi so I think in this case using SAMA5_FUSE could be misleading.
Whether you create SAMA5_FUSE and SAMA5_SEC_FUSE to differentiate them or you keep SAMA5_FUSE for both, but create a subtype: SAMA5_FUSE_ORDINARY and SAMA5_FUSE_SECURITY

Reply:
I'll take a look and rework it :)

Reply:
Nevermind, I missed the SAMA5_HAVE_SFC on `config SAMA5_SFC`. You implementation is fine, but we need to enable to SAMA5_HAVE_FUSE if ARCH_CHIP_SAMA5D3 as you noticed

Reply:
Thank you for merging - my best attempt so far with no changes, Except...

@acassis 
> ... but we need to enable to SAMA5_HAVE_FUSE if ARCH_CHIP_SAMA5D3 as you noticed

@xiaoxiang781216 
> ...why not move the private macro/variable to source file

I will submit a small PR next week to correct these 2 minor things :)